staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta_test.go

```diff
@@ -498,3 +498,65 @@ func TestValidateAnnotations(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateImmutableField(t *testing.T) {
```

Comment:
Most of these tests are actually testing that DeepEqual works, I suspect you could simplify, but that's also probably fine.